src/workerd/api/kv-test.js

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import assert from 'node:assert';
+export default {
+  // Producer receiver (from `env.NAMESPACE`)
+  async fetch(request, env, ctx) {
+    const options = {
+      status: 200,
+      statusText: "Success!",
+      headers: new Headers({
+        'Content-Type': 'application/json'
+      })
+    };
+
+    var result = "example";
+    const { pathname } = new URL(request.url);
+    if (pathname === '/fail-client') {
+      options.status = "404"
+      result = ""
+    } else if (pathname == "/fail-server") {
+      options.status = "500"
+      result = ""
+    } else if (pathname == "/get-json") {
+      result = JSON.stringify({ example: "values" });
+    } else if (pathname == "/bulk/get") {
+      var body = request.body;
+      const reader = body.getReader();
+      const decoder = new TextDecoder(); // UTF-8 by default
+      let r = "";
+      while (true) {
+          const { done, value } = await reader.read();
+          if (done) break;
+          r += decoder.decode(value, { stream: true });
+      }
+      const parsedBody = JSON.parse(r);
+      const keys = parsedBody.keys;
+      if (keys.length < 1) {
+        return new Response("Array needs at least one key", {status: 400})
+      }
+      result = {}
+      if(parsedBody.type == "json") {
+        for(const key of keys) {
+          if(key == "key-not-json") {
+            return new Response(null, {status: 500})
```

Comment:
fwiw, We have `Response.error()` now that would end up sending a 500 response back to the eyeball.